forge-gui/res/cardsfolder/m/manifold_insights.txt

```diff
@@ -2,7 +2,7 @@ Name:Manifold Insights
 ManaCost:2 U
 Types:Sorcery
 A:SP$ PeekAndReveal | PeekAmount$ 10 | RevealValid$ Card | RememberRevealed$ True | SubAbility$ ChooseNonLand | SpellDescription$ Reveal the top ten cards of your library. Starting with the next opponent in turn order, each opponent chooses a different nonland card from among them. Put the chosen cards into your hand and the rest on the bottom of your library in a random order.
-SVar:ChooseNonLand:DB$ RepeatEach | RepeatPlayers$ Player.Opponent | RepeatSubAbility$ OpponentsChoose | SubAbility$ ShipToBottom
+SVar:ChooseNonLand:DB$ RepeatEach | RepeatPlayers$ Player.Opponent | InTurnOrder$ True | RepeatSubAbility$ OpponentsChoose | SubAbility$ ShipToBottom
```

Comment:
here the extra param wouldn't work yet since the RepeatPlayers one uses a different syntax

I think what needs to happen instead is RepeatEachEffect calling `getDefinedPlayersOrTargeted` because there is already some logic to sort the players:

https://github.com/Card-Forge/forge/blob/4fcd6f577acffa7d64e0e99437bfe36896953b7b/forge-game/src/main/java/forge/game/ability/SpellAbilityEffect.java#L303-L306

so this just needs some fixing to make it work even if the AP isn't included

Reply:
Understood. I really should have looked more into this individual card before committing a change for it.

Reply:
no problem, all easy cards for this set have been finished for a while, so now this helps improving the engine :)